### PR TITLE
Add CPython 3.8.20, 3.9.20, 3.10.15, 3.11.10 and 3.12.6

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Make sure you have checked all steps below.

### Prerequisite
* [ ] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
* [ ] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project.
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.
* [ ] My PR addresses the following pyenv issue (if any)
  - Closes https://github.com/pyenv/pyenv/issues/XXXX

### Description
- [ ] Here are some details about my PR

### Tests
- [ ] My PR adds the following unit tests (if any)

### CODE REVIEW

File: plugins/python-build/share/python-build/3.12.6 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+prefer_openssl3
+export PYTHON_BUILD_CONFIGURE_WITH_OPENSSL=1
+install_package "openssl-3.3.2" "https://www.openssl.org/source/openssl-3.3.1.tar.gz#2e8a40b01979afe8be0bbfb3de5dc1c6709fedb46d6c89c10da114ab5fc3d281" mac_openssl --if has_broken_mac_openssl
```
Comment (jsirois) [inline plugins/python-build/share/python-build/3.12.6:3]: Did you mean `https://github.com/openssl/openssl/releases/download/openssl-3.3.2/openssl-3.3.2.tar.gz#2e8a40b01979afe8be0bbfb3de5dc1c6709fedb46d6c89c10da114ab5fc3d281` or should `install_package "openssl-3.3.2"` be `install_package "openssl-3.3.1"`?

Comment (edgarrmondragon) [inline plugins/python-build/share/python-build/3.12.6:3]:
```suggestion
install_package "openssl-3.3.2" "https://www.openssl.org/source/openssl-3.3.2.tar.gz#2e8a40b01979afe8be0bbfb3de5dc1c6709fedb46d6c89c10da114ab5fc3d281" mac_openssl --if has_broken_mac_openssl
```

**Comment by jeremy** [comment]
Noting that https://www.openssl.org/source/openssl-3.3.2.tar.gz is 404ing and won't be supported: https://github.com/openssl/openssl/issues/25376

References https://github.com/jdx/mise/issues/2552

**Comment by edgarrmondragon** [comment]
> Noting that https://www.openssl.org/source/openssl-3.3.2.tar.gz is 404ing and won't be supported: [openssl/openssl#25376](https://github.com/openssl/openssl/issues/25376)
> 
> References [jdx/mise#2552](https://github.com/jdx/mise/issues/2552)

@jeremy Care to open a PR to downgrade to 3.3.1?

**Comment by jeremy** [comment]
Need to switch to the new OpenSSL release download URLs. (Just a drive-by to resolve the mise issue that tickled this; I'm not a user of pyenv myself.)

**Comment by edgarrmondragon** [comment]
> Need to switch to the new OpenSSL release download URLs. (Just a drive-by to resolve the mise issue that tickled this; I'm not a user of pyenv myself.)

Opened https://github.com/pyenv/pyenv/pull/3059

**Pull request merged**